kube-runtime/src/scheduler.rs

```diff
@@ -126,6 +126,53 @@ impl<'a, T: Hash + Eq + Clone, R> SchedulerProj<'a, T, R> {
             }
         }
     }
+
+    /// Attempt to retrieve a message from queue and mark it as pending.
+    pub fn pop_queue_message_into_pending(&mut self, cx: &mut Context<'_>) -> Poll<T> {
+        loop {
```

Comment:
could this be a `while let Poll::Ready(Some(msg)) = self.queue.poll_expired(cx) {` ?